src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -171,14 +172,21 @@ await Task.Delay(_smartRenameSession.AutomaticFetchDelay, cancellationToken)
         {
             var document = this.BaseViewModel.Session.TriggerDocument;
             var smartRenameContext = ImmutableDictionary<string, string[]>.Empty;
-            var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
-            var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
-                .ConfigureAwait(false);
-            var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
-                .ConfigureAwait(false);
-            smartRenameContext = ImmutableDictionary.CreateRange<string, string[]>(
-                context
-                .Select(n => new KeyValuePair<string, string[]>(n.Key, n.Value.ToArray())));
+            try
+            {
+                var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+                var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
+                    .ConfigureAwait(false);
+                var context = await editorRenameService.GetRenameContextAsync(this.BaseViewModel.Session.RenameInfo, renameLocations, cancellationToken)
+                    .ConfigureAwait(false);
+                smartRenameContext = ImmutableDictionary.CreateRange<string, string[]>(
+                    context
+                    .Select(n => new KeyValuePair<string, string[]>(n.Key, n.Value.ToArray())));
+            }
+            catch (Exception e) when (FatalError.ReportAndCatch(e, ErrorSeverity.Diagnostic))
```

Comment:
I tried both `ErrorSeverity.Diagnostic` and `ErrorSeverity.General` but did not see any new telemetry. What do you recommend I use here?

Reply:
`Diagnostic` should be OK.
I think it should be classified as NonFatalWaston event